### PR TITLE
Cli milestone part-1

### PR DESCRIPTION
**Pull request by Dmaddu**
### :athletic_shoe: How to Build and Test the Change

1. check out the repo
2. uncomment the lines from 111 to 151 in components/automate-cli/cmd/chef-automate/upgrade.go
3. build automate-cli, automate-deployment, automate-ui 
4. start all the services(start_all_services)
5. once automate is up and running, run `chef-automate upgrade status`
6. run `chef-automate upgrade run --major`

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).
With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

**Review by kalroy — CHANGES_REQUESTED** [review]

File: components/automate-cli/cmd/chef-automate/upgrade.go [hunk 1]
```diff
@@ -96,8 +107,53 @@ func runUpgradeCmd(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+
```
[the remark this answers is not in the document]
Comment (Dmaddu) [reply components/automate-cli/cmd/chef-automate/upgrade.go:110]: @kalroy  This commented code will be uncommented in part-2.

File: components/automate-ui/src/app/ui.component.ts [hunk 2]
```diff
@@ -77,5 +80,32 @@ export class UIComponent implements OnInit, AfterViewChecked {
     if (this.chefSessionService.connector && this.userPrefsService.uiSettings.isTimeformatExist) {
       this.store.dispatch(new GetUserPreferences());
     }
+
+    // manual upgrade banner
+    const bannerStorage = localStorage.getItem('manual-upgrade-banner');
+    if (isNull(bannerStorage)) {
+      localStorage.setItem('manual-upgrade-banner', this.booleanToString(true));
```
Comment (vivek-yadav) [inline components/automate-ui/src/app/ui.component.ts:87]: I think this also should be set as 
```ts
localStorage.setItem('manual-upgrade-banner', this.booleanToString(false));
```

File: ec2/Vagrantfile [hunk 3]
```diff
@@ -100,7 +100,8 @@ if ['up', 'ssh', 'destroy', 'halt'].include?(ARGV[0])
         raise "#{ENV['HOME']}/.okta_aws.toml is not defined, cannot continue. Please read README.md for an example and usage details."
       end
       puts " * You might be prompted for your Okta password now..."
-      `okta_aws "#{aws_profile}"`
+      #`okta_aws "#{aws_profile}"`
+      `saml2aws login --profile chef-engineering`
```
Comment (vivek-yadav) [inline ec2/Vagrantfile:104]: why is this change needed in Milestone Version work ?

Comment (Dmaddu) [reply ec2/Vagrantfile:104]: technically, these changes no way related to milestone changes, while setting up the ec2 for testing, we modified this file as above, and since these changes are needed for setting up ec2 using vagrant, we haven't reverted the changes.

File: components/automate-ui/src/app/ui.component.ts (hunk 2 above)
Comment (Dmaddu) [reply components/automate-ui/src/app/ui.component.ts:87]: This is the current behavior, and to minimize the changes to enable the banner I feel the change is not quired.

https://user-images.githubusercontent.com/44021874/157400341-4ff422f2-312d-4327-b55e-7acf05039d33.mov

**Review by kalroy — APPROVED** [review]

**Review by vivekshankar1 — APPROVED** [review]

File: components/automate-deployment/pkg/server/upgrade_status.go [hunk 4]
```diff
@@ -24,35 +26,68 @@ import (
 // I think this should get a lot more simple as we integrate the
 // converger. In that world we can potentially have a function that
 // takes an expected state and tells us if it is true or not.
```
Comment (shivangi-singh1) [inline components/automate-deployment/pkg/server/upgrade_status.go:28]: We may remove this comment if not needed.

File: ec2/Vagrantfile [hunk 5]
```diff
@@ -100,7 +100,8 @@ if ['up', 'ssh', 'destroy', 'halt'].include?(ARGV[0])
         raise "#{ENV['HOME']}/.okta_aws.toml is not defined, cannot continue. Please read README.md for an example and usage details."
       end
       puts " * You might be prompted for your Okta password now..."
-      `okta_aws "#{aws_profile}"`
+      #`okta_aws "#{aws_profile}"`
```
Comment (shivangi-singh1) [inline ec2/Vagrantfile:103]: Please remove commented code.

File: components/automate-deployment/pkg/server/upgrade_status.go (hunk 4 above)
Comment (Dmaddu) [reply components/automate-deployment/pkg/server/upgrade_status.go:28]: this comment is needed for reference.

File: ec2/Vagrantfile (hunk 5 above)
Comment (Dmaddu) [reply ec2/Vagrantfile:103]: Done

**Review by shaik80 — APPROVED** [review]

**Review by shivangi-singh1 — APPROVED** [review]

File: components/automate-ui/src/app/ui.component.ts (hunk 2 above)
Comment (Dmaddu) [reply components/automate-ui/src/app/ui.component.ts:87]: @vivek-yadav  updated the code

**Review by vivek-yadav — APPROVED** [review]
